website/content/specification/_index.md

```diff
@@ -1,50 +1,1320 @@
+
+In XML, the value of a field appears as the text child of the field's element. In JSON and YAML, a property name is needed for the value. The `<json-value-key>` and `<json-value-key-flag>` elements provide a means to control the behavior of how this value is represented.
+
+If no `<json-value-key>` or `<json-value-key-flag>` element is declared, a property name value will be chosen based on the data type as follows:
+  - If the field's `@as-type` is `markup-line`, then the property name will be `RICHTEXT`.
+  - If the field's `@as-type` is `markup-multiline`, then the property name will be `prose`.
+  - Otherwise, the property name will be `STRVALUE`.
+
+This logic may result in less than ideal property names. Metaschema provides the <json-value-key>` and `<json-value-key-flag>` elements to override this behavior. Use of these elements are mutually exclusive.
+
+The following subsections describe the use of these elements.
+
+#### `<json-value-key>`
+
+The `<json-value-key>` element can be declared to set the property name for the field's value. Its text value child MUST be used as the property name for the field's value.
+
+For example:
+
+```xml {linenos=table,hl_lines=[5]}
+<define-assembly name="assembly">
+  <root-name>assembly</root-name>
+  <model>
+    <define-field name="field" max-occurs="unbounded">
+      <json-value-key>value</json-value-key>
+      <group-as name="fields" in-json="ARRAY"/>
+      <define-flag name="flag-default-a" default="a"/>
+    </define-field>
+  </model>
+</define-assembly>
+```
+
+Would allow the following content in JSON and YAML.
+
+{{< tabs JSON YAML >}}
+{{% tab %}}
+```json {linenos=table,hl_lines=[6]}
+{
+  "assembly": {
+    "fields": [
+      {
+        "flag-default-a": "b",
+        "value": "value1"
+      }
+    ]
+  }
+}
+```
+{{% /tab %}}
+{{% tab %}}
+```yaml {linenos=table,hl_lines=[5]}
+---
+assembly:
+  fields:
+  - flag-default-a: "b"
+    value: "value1"
+```
+{{% /tab %}}
+{{% /tabs %}}
+
+#### `<json-value-key-flag>`
+
+The `<json-value-key>` element can be declared to use the value of specific flag on the field as the property name for the field's value.
+
+The `@flag-name` attribute MUST reference the [name](#name) of a flag on the field whose value is to be used as the property name.
+
+This results in a property that is a combination of the referenced flag's value and the field's value.
+
+For example:
+
+TODO: complete this example.
+
+## top-level `<define-assembly>`
+
+An assembly definition, represented by the `<define-assembly>` element, is used to declare a reusable [assembly](/specification/terminology/#assembly) within a Metaschema module.
+
+An assembly definition provides the means to implement a complex, composite, named [*information element*](/specification/terminology/#information-element) that has no value. An assembly definition consists of an optional set of [*flags*](#flag) and an optional sequence of model instances, which are instances of assemblies and fields.
```

Comment:
```suggestion
An assembly definition provides the means to implement a complex, composite, named [*information element*](/specification/terminology/#information-element), with no value of its own, used to collect and organize other information elements. An assembly definition comprises a set of [*flags*](#flag), optional or required, and a sequence of model instances, i.e. instances of assemblies and fields, which may be optional, required or repeatable.
```

Reply:
Not a fan of *comprises* as it makes it read the a set of flags are made up of an assembly. Flags can't be both optional and required.

How about the following.

```suggestion
An assembly definition provides the means to implement a complex, composite, named [*information element*](/specification/terminology/#information-element) that collects and organizes other information elements, with no value of its own. 

An assembly definition consists of an optional set of [*flags*](#flag) and an optional sequence of [model instances](#model-instances).
```
The model instance section can cover the remaining details about model instances.